client/src/components/editBlog/edit.js

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import FormErrors from '../errors/formErrors';
+import Constants from '../../helpers/constants';
+import FormElement from '../formElements/formElement';
+import './edit.css';
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux';
+import * as blogActions from '../../store/actions/blog'
+import Loader from 'react-loader'
+
+class Edit extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      blog: {
+        title: '',
+        author: '',
+        body: ''
+      },
+      formErrors: { title: '', author: '', body: '' },
+      titleValid: true,
+      authorValid: true,
+      bodyValid: true,
+      formValid: true
+    };
+  }
+
+  static propTypes = {
+    getBlog: PropTypes.func.isRequired,
+    blog: PropTypes.object.isRequired
+  }
+
+  static defaultProps = {
+    blog: {}
+  }
+
+  componentDidMount() {
+    const { id } = this.props.match.params;
+    this.props.getBlog(id);
+  }
+
+  onStateUpdated = (name, value) => { this.validateField(name, value) };
+
+  onChange = (e) => {
+    const { name, value } = e.target;
+    const state = this.props.blog
+    state[name] = value;
+    this.setState({ blog: state },
+      this.onStateUpdated(name, value));
+  }
+
+  onSubmit = (e) => {
+    e.preventDefault();
```

Comment:
What is the purpose of this line tell me, pls?

Reply:
It prevents form submittion and does all other necessary job